tests/unit/s2n_connection_test.c

```diff
@@ -245,20 +245,22 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_get_selected_digest_algorithm(conn, &output));
         EXPECT_EQUAL(S2N_TLS_HASH_NONE, output);
 
-        s2n_tls_hash_algorithm expected_output[] = {
-            S2N_TLS_HASH_NONE, S2N_TLS_HASH_MD5,
-            S2N_TLS_HASH_SHA1, S2N_TLS_HASH_SHA224,
-            S2N_TLS_HASH_SHA256, S2N_TLS_HASH_SHA384,
-            S2N_TLS_HASH_SHA512, S2N_TLS_HASH_MD5_SHA1,
-            S2N_TLS_HASH_NONE
+        s2n_tls_hash_algorithm expected_output[S2N_HASH_ALGS_COUNT] = {
+            [S2N_HASH_MD5] = S2N_TLS_HASH_MD5,
+            [S2N_HASH_SHA1] = S2N_TLS_HASH_SHA1,
+            [S2N_HASH_SHA224] = S2N_TLS_HASH_SHA224,
+            [S2N_HASH_SHA256] = S2N_TLS_HASH_SHA256,
+            [S2N_HASH_SHA384] = S2N_TLS_HASH_SHA384,
+            [S2N_HASH_SHA512] = S2N_TLS_HASH_SHA512,
+            [S2N_HASH_MD5_SHA1] = S2N_TLS_HASH_MD5_SHA1,
         };
 
         for (size_t i = S2N_TLS_HASH_NONE; i <= UINT16_MAX; i++) {
             struct s2n_signature_scheme test_scheme = *conn->handshake_params.client_cert_sig_scheme;
             test_scheme.hash_alg = i;
             conn->handshake_params.client_cert_sig_scheme = &test_scheme;
             conn->handshake_params.server_cert_sig_scheme = &test_scheme;
-            if (i <= S2N_HASH_ALGS_COUNT) {
+            if (i < S2N_HASH_ALGS_COUNT) {
```

Comment:
This should be a familiar carry-over from my previous approach: https://github.com/aws/s2n-tls/pull/5262#discussion_r2053100614